ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoProvider.java

```diff
@@ -67,74 +69,97 @@ public SearchTreeRaoProvider() {
-            CracResultUtil.applyRemedialActionsForState(raoInput.getNetwork(), raoInput.getCrac(), preventiveRaoResult.getPostOptimVariantId(), raoInput.getCrac().getPreventiveState());
-            Map<State, RaoResult> curativeResults = optimizeCurativePerimeters(raoInput, parameters, curativeTreeParameters, network);
+        // merge variants
+        LOGGER.info("Merging preventive and curative RAO results.");
+        RaoResult mergedRaoResults = mergeRaoResults(raoInput.getCrac(), preventiveRaoResult, curativeResults);
 
-            LOGGER.info("Merging preventive and curative RAO results.");
-            RaoResult mergedRaoResults = mergeRaoResults(raoInput.getCrac(), preventiveRaoResult, curativeResults);
-            // For logs only
-            if (mergedRaoResults.isSuccessful()) {
-                boolean relativePositiveMargins =
-                        parameters.getObjectiveFunction().equals(RaoParameters.ObjectiveFunction.MAX_MIN_RELATIVE_MARGIN_IN_AMPERE) ||
-                                parameters.getObjectiveFunction().equals(RaoParameters.ObjectiveFunction.MAX_MIN_RELATIVE_MARGIN_IN_MEGAWATT);
-                SearchTreeRaoLogger.logMostLimitingElementsResults(raoInput.getCrac().getBranchCnecs(), mergedRaoResults.getPostOptimVariantId(), parameters.getObjectiveFunction().getUnit(), relativePositiveMargins);
-            }
-            return CompletableFuture.completedFuture(mergedRaoResults);
+        // log results
+        if (mergedRaoResults.isSuccessful()) {
+            boolean relativePositiveMargins =
+                parameters.getObjectiveFunction().equals(RaoParameters.ObjectiveFunction.MAX_MIN_RELATIVE_MARGIN_IN_AMPERE) ||
+                    parameters.getObjectiveFunction().equals(RaoParameters.ObjectiveFunction.MAX_MIN_RELATIVE_MARGIN_IN_MEGAWATT);
+            SearchTreeRaoLogger.logMostLimitingElementsResults(raoInput.getCrac().getBranchCnecs(), mergedRaoResults.getPostOptimVariantId(), parameters.getObjectiveFunction().getUnit(), relativePositiveMargins);
         }
+        return CompletableFuture.completedFuture(mergedRaoResults);
+
+    }
+
+    private CompletableFuture<RaoResult> optimizeOneStateOnly(RaoInput raoInput, RaoParameters raoParameters) {
+        RaoData raoData = new  RaoData(
+            raoInput.getNetwork(),
+            raoInput.getCrac(),
+            raoInput.getOptimizedState(),
+            raoInput.getPerimeter(),
+            raoInput.getReferenceProgram(),
+            raoInput.getGlskProvider(),
+            raoInput.getBaseCracVariantId(),
+            raoParameters);
+        TreeParameters treeParameters = TreeParameters.buildForPreventivePerimeter(raoParameters.getExtension(SearchTreeRaoParameters.class));
+        new InitialSensitivityAnalysis(raoData).run();
+        return CompletableFuture.completedFuture(new SearchTree().run(raoData, raoParameters, treeParameters).join());
```

Comment:
why .join() ?

Reply:
Don't know xD, it was indeed quite strange. I replace it with 
`         return new SearchTree().run(raoData, raoParameters, treeParameters); `